### ISSUE TITLE
Feature Request: Watch Mode

### ISSUE THREAD

**Issue by albohlabs** [dmmulroy/tsc.nvim#22]
Hi, 

this is a really nice plugin. Saved me a some time in my workflow. Thanks a lot for that.

Have you thought about adding a watch mode with a corresponding TSCWatch command? Do you see any problems with this?

**Comment by dmmulroy** [comment]
I think this is good idea and I've given a little bit of thought to it. I need to learn more about how `vim.fn.jobstart` works. Right now the limiting factor to a "real" watch mode is this option passed to `jobstart`: `stdout_buffered: (boolean) Collect data until EOF (stream closed) before invoking 'on_stdout'. `

So the plugin currently waits until the stream is closed from the child process to start parsing errors. I'll need to play around and think about how to keep the process alive and what data looks like unbuffered. 

That being said, a quick and easy workaround would be to use an autocommand (which maybe I'll introduce as an initial version of this)


```lua
vim.api.nvim_create_autocmd("BufWritePre", {
    group = vim.api.nvim_create_augroup("tsc_on_save", { clear = true }),
    pattern = "*.ts",
    desc = "Run TSC on save",
    callback = function()
        require('tsc').run()
    end,
})
```

**Comment by dmmulroy** [comment]
Dropping this here so I don't forget about it - might be able to use rpc

```
You can also embed Nvim via |jobstart()|, and communicate using |rpcrequest()|
and |rpcnotify()|:
>vim
    let nvim = jobstart(['nvim', '--embed'], {'rpc': v:true})
    echo rpcrequest(nvim, 'nvim_eval', '"Hello " . "world!"')
    call jobstop(nvim)
```

**Comment by albohlabs** [comment]
This means: Consume the output of `tsc -w` cmd via an rpc `on_stdout` callback and read it line by line into a buffer. There is an example of how to do this on the https://neovim.io/doc/user/channel.html#channel-bytes web site.

```lua
let s:lines = ['']
func! s:on_event(job_id, data, event) dict
  let eof = (a:data == [''])
  " Complete the previous line.
  let s:lines[-1] .= a:data[0]
  " Append (last item may be a partial line, until EOF).
  call extend(s:lines, a:data[1:])
endf
``` 
But there is no EOF character. So the output should be continuously checked for something like `[8:06:02 PM] Found 2 errors. Watching for file changes.

**Comment by albohlabs** [comment]
Vgit searches for a `\n' in a constant stream of stdout. https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/core/ReadStream.lua

**Comment by dmmulroy** [comment]
Resolved by #34
